tracee/event_monitor_ebpf.c

```diff
@@ -453,15 +455,28 @@ static __always_inline int get_config(u32 key)
 static __always_inline int should_trace()
 {
     struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+    int config_mode = get_config(CONFIG_MODE);
+    u32 rc = 0;
+    u32 host_pid = bpf_get_current_pid_tgid() >> 32;
 
-    if (get_config(CONFIG_TRACEE_PID) == bpf_get_current_pid_tgid() >> 32)
+    if (get_config(CONFIG_TRACEE_PID) == host_pid)
         return 0;
 
-    u32 rc = 0;
-    if (get_config(CONFIG_MODE) == MODE_CONTAINER)
+    // All logs all processes except tracee itself
+    if (config_mode == MODE_PROCESS_ALL)
+        return 1;
+    else if (config_mode == MODE_CONTAINER_NEW)
         rc = lookup_pid_ns(task);
-    else
+    else if (config_mode == MODE_PROCESS_NEW || config_mode == MODE_PROCESS_LIST)
         rc = lookup_pid();
+    else if (config_mode == MODE_CONTAINER_ALL) {
+        // 'Container-All' means anything in a container
+        // We can check if we're in a namespace by checking
+        // our PID Vs 'real' PID on host
+        if (get_task_ns_tgid(task) != host_pid) {
```

Comment:
@yanivagman I draw your attention to this semi-hacky way to check if a process is running in a container or not, as `MODE_CONTAINER_ALL` really means 'any process thats in a container'. Are you ok with this assumtion?

Reply:
Yes, I'm ok with it. 
As kernel isn't aware about containers, but rather works with namespaces and cgroups, there is no other way than using these kind of hacks. For example, I used "pid == 1" in the code to know if a new container started (the assumption here is that a container will use a new pid namespace, which is not always the case).

In kernel 4.18, new bpf helpers were added that allow to better work with cgroups (e.g. bpf_get_current_cgroup_id()) - we may change the way we identify containers in the future to work with these helpers